packages/sanity/src/core/releases/tool/detail/documentTable/extractSearchableText.ts

```diff
@@ -0,0 +1,52 @@
+import {toPlainText} from '@portabletext/react'
+import {isPortableTextBlock} from '@portabletext/toolkit'
+
+/**
+ * Extracts a searchable string from a field value, handling various data types
+ * @param value - The field value to extract text from
+ * @returns A string representation suitable for search
+ */
+export function extractSearchableText(value: unknown): string {
+  if (typeof value === 'undefined' || value === null) return ''
+
+  // Handle primitive types
+  if (typeof value === 'string') return value
+  if (typeof value === 'number') return value.toString()
+
+  // Handle arrays
+  if (Array.isArray(value)) {
+    // Check if it's portable text
+    if (value.length > 0 && isPortableTextBlock(value[0])) {
+      return toPlainText(value)
+    }
+    // Handle regular arrays (join with space)
+    return value.map(extractSearchableText).filter(Boolean).join(' ')
+  }
+
+  // Handle objects
+  if (typeof value === 'object' && value !== null) {
+    // Handle localized objects (common pattern: {en: "title", no: "tittel"})
+    const entries = Object.entries(value)
+    if (
+      entries.length > 0 &&
+      entries.every(([key, val]) => typeof val === 'string' && !key.startsWith('_'))
+    ) {
+      // This looks like a localized object, extract all values (excluding metadata keys)
+      return entries.map(([, val]) => val).join(' ')
+    }
+
+    // Handle objects with specific properties (like {title: "value"})
+    if ('title' in value && typeof value.title === 'string') return value.title
```

Comment:
what if it has both title, name and text, don't we want to extract all of them?

Reply:
🤔 
_technically_ I suppose, yes. The way that I've been approaching this for the most part is that the title is the "primary" source to be concerned about and the rest are more fallbacks? 🤔 
But I can instead make this just grab all of them? Do you have any thoughts?

Reply:
Actually we do extract all of them in other instances, so let me fix that as well